pyspec/msdial_msms/msdial_msms.py

```diff
@@ -0,0 +1,354 @@
+    -------
+    bool
+        True if cell value == 'Average Rt(min)', otherwise False
+    """
+    
+    return sheet.cell(row = 4, column = 2).value == 'Average Rt(min)'
+
+def mzColumn(sheet):
+    """Returns True if row 4 and column 3 of excel sheet value is 'Average Mz'
+            
+    Parameters
+    ----------
+    sheet : excel workbook sheet
+        The first sheet in excel workbook opened
+            
+    Returns
+    -------
+    bool
+        True if cell value == 'Average Mz', otherwise False
+    """
+    
+    return sheet.cell(row = 4, column = 3).value == 'Average Mz'
+
+def msmsColumn(sheet):
+    """Returns True if row 4 and column 3 of excel sheet value is 'MS/MS spectrum'
+            
+    Parameters
+    ----------
+    sheet : excel workbook sheet
+        The first sheet in excel workbook opened
+            
+    Returns
+    -------
+    bool
+        True if cell value == 'MS/MS spectrum', otherwise False
+    """
+    
+    return sheet.cell(row = 4, column = 22).value == 'MS/MS spectrum'
+    
+def test_retentionTimeColumn(sheet):
+    """py_test to make sure retention time column is correct in excel file opened.
+            
+    Parameters
+    ----------
+    sheet : excel workbook sheet
+        The first sheet in excel workbook opened
+    """
+    
+    assert retentionTimeColumn(sheet) == True
+    
+def test_mzColumn(sheet):
+    """py_test to make sure mass to charge column is correct in excel file opened.
+            
+    Parameters
+    ----------
+    sheet : excel workbook sheet
+        The first sheet in excel workbook opened
+    """
+    
+    assert mzColumn(sheet) == True
+    
+def test_msmsColumn(sheet):
+    """py_test to make sure msms column is correct in excel file opened.
+            
+    Parameters
+    ----------
+    sheet : excel workbook sheet
+        The first sheet in excel workbook opened
+    """
+    
+    assert msmsColumn(sheet) == True
+
+def collectFeatures(sheet, filename):
+    """For all rows with MSMS data, stores feature objects in feature tuple
+            
+    Parameters
+    ----------
+    sheet : excel workbook sheet
+        The first sheet in excel workbook opened
+    filename : str
+        The file name of the excel document being worked with
+            
+    Returns
+    -------
+    tuple
+        Tuple of class Feature objects
+    """
+    
+    features = ()#initiate empty tuple
+    #count = 0
+    current_row = 5
+    
+    #py_tests to check excel sheet is correct format
+    test_retentionTimeColumn(sheet)
+    test_mzColumn(sheet)
+    test_msmsColumn(sheet)
+    
+    #check all rows for msms data and store data if msms data is present
+    while (current_row <= sheet.max_row):
+        if msmsFound(sheet, current_row):
+            feature = Feature(sheet, current_row)
+            features = features + (feature,)
+            #count = count + 1
+        
+    print(str(len(features)) + ' features with MSMS found in ' + filename)
```

Comment:
please use python logging. Print can easily crash systems, if you write to stderr and sdtout at the same time. This can be quite often observed in PyCharm